pkg/controller/job/job_controller.go

```diff
@@ -333,7 +336,12 @@ func (jm *Controller) updatePod(old, cur interface{}) {
 		return
 	}
 
-	// Otherwise, it's an orphan. If anything changed, sync matching controllers
+	// Otherwise, it's an orphan.
+	// Clean the finalizer.
+	if hasJobTrackingFinalizer(curPod) {
+		jm.enqueueOrphanPod(curPod)
```

Comment:
same question about async orphan pod handling removing the finalizer even if we add an ownerRef in the meantime?